test/qa_selector.cpp

```diff
@@ -0,0 +1,320 @@
+        }
+
+        std::unordered_map<std::size_t, std::size_t> used_inputs;
+        for (const auto &[input_index, output_index] : _internalMapping) {
+            auto *input_reader    = ins[input_index];
+            auto *output_writer   = outs[output_index];
+
+            auto  input_available = [&] {
+                auto it = used_inputs.find(input_index);
+                if (it != used_inputs.end()) {
+                    return it->second;
+                }
+
+                const auto available     = input_reader->available();
+                used_inputs[input_index] = available;
+                return available;
+            }();
+
+            auto input_span = input_reader->get(input_available);
+
+            if (input_available > 0) {
+                auto output_span = output_writer->reserve_output_range(input_available);
+
+                for (std::size_t i = 0; i < input_span.size(); ++i) {
+                    output_span[i] = input_span[i];
+                }
+
+                output_span.publish(input_available);
+            }
+        }
+
+        if (const auto select_available = select->available(); select_available > 0) {
+            auto select_span = select->get(select_available);
+            _selectedSrc     = select_span.back();
+
+            if (_selectedSrc >= 0 && _selectedSrc < ins.size()) {
+                // write to optional fixed monitor output
+                // auto monitor_written_count = ins[_selectedSrc]->available();
+                // std::ranges::copy(monOut, ins[_selectedSrc]);
+                // const std::size_t publishedSamples = std::min(monOut.size(), ins[_selectedSrc].size());
+                // monOut.publish(publishedSamples);
+            }
+
+            select->consume(select_available); // consume all samples on the 'select' streaming input port
+        }
+
+        for (auto src_port = 0U; src_port < ins.size(); ++src_port) {
+            if (auto it = used_inputs.find(src_port); it != used_inputs.end()) {
+                // If we read from this input, consume exactly the number of bytes we read
+                ins[src_port]->consume(it->second);
+
+            } else if (backPressure) {
+                ins[src_port]->consume(0_UZ);
+
+            } else {
+                // make the implicit consume all available behaviour explicit
+                ins[src_port]->consume(ins[src_port]->available());
+            }
+        }
+
+        // N.B. some corner case-handling to be added:
+        // * one input mapped to multiple outputs -> check for consistent and produce the same min available() for all mapped outputs
+
+        return fg::work_return_status_t::OK;
+    }
+};
+} // namespace gr::blocks::basic
+
+ENABLE_REFLECTION_FOR_TEMPLATE_FULL((typename T), (gr::blocks::basic::Selector<T>), selectOut, inputs, monitorOut, outputs, nInputs, nOutputs, mapIn, mapOut, backPressure);
+static_assert(fg::HasProcessBulkFunction<gr::blocks::basic::Selector<double>>);
```

Comment:
Yes, I thought of that as well. And I'd add some basic checks to the macro so that we can verify as much as possible that a block is correctly implemented.